File: task-03/Python/Sub task 3/subtask3.py
def print_diamond(n):
    if n < 1:
        print("Number should be greater than 1")
        return

    mid = n // 2

    # Upper part of the diamond
    for i in range(1, n + 1, 2):
        # Calculate and print leading spaces
        spaces = mid - i // 2
        print(" " * spaces + "*" * i)

    # Lower part of the diamond
    if n % 2 == 0:
        for i in range(n - 1, 0, -2):
            # Calculate and print leading spaces with a slight shift for asymmetry
            spaces = mid - i // 2 + 1
            print(" " * (spaces - 1) + "*" * i)
    else:
        for i in range(n - 2, 0, -2):
            spaces = mid - i // 2
            print(" " * spaces + "*" * i)

# Example usage
if __name__ == "__main__":
    n = int(input("Enter the number of rows for the diamond: "))
    print_diamond(n)

File: task-03/Java/Sub task 3/subtask3.js
function printDiamond(n) {
    if (n < 1) {
        console.log("Number should be greater than 1");
        return;
    }

    const mid = Math.floor(n / 2);

    // Upper part of the diamond
    for (let i = 1; i <= n; i += 2) {
        // Calculate and print leading spaces
        const spaces = mid - Math.floor(i / 2);
        console.log(" ".repeat(spaces) + "*".repeat(i));
    }

    // Lower part of the diamond
    if (n % 2 === 0) {
        for (let i = n - 1; i >= 1; i -= 2) {
            // Calculate and print leading spaces with a slight shift for asymmetry
            const spaces = mid - Math.floor(i / 2) + 1;
            console.log(" ".repeat(spaces - 1) + "*".repeat(i));
        }
    } else {
        for (let i = n - 2; i >= 1; i -= 2) {
            const spaces = mid - Math.floor(i / 2);
            console.log(" ".repeat(spaces) + "*".repeat(i));
        }
    }
}

// Example usage
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question("Enter the number of rows for the diamond: ", n => {
    printDiamond(parseInt(n));
    readline.close();
});

